### PR TITLE
Issue/8510 site creation accessibility

### LINKED ISSUE #8510: [Site creation] Accessibility review
This issue is intended to capture the need to preserve proper accessibility support with the addition of both new screens & UI elements introduced when enhancing site creation.

Check the flow with
- TalkBack
- Switch Access
- Accessibility Scanner

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #8510 

~~**Not Ready for Review** -> I'll ping you when it's ready  (`issue/site-creation-design-review` needs to be merged first).~~

I've tried to fix all issues which came up when using TalkBack and Switch Control. I've also checked the app using the Accessibility Scanner - there are couple of minor issues which I decided are not worth fixing (contrast is for example 4.29 instead of 4.5).

Test the Site Creation flow with
1. Talkback
2. Switch Access

Update release notes:
- [ X ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

@khaykov Could you please review this PR as you are most familiar with all the possible accessibility issues. Thanks!

### CODE REVIEW

**Comment by khaykov** [comment]
Thanks for the PR, @malinajirka !

I have a couple of comments regarding accessibility :)

EditText at steps 1,2 and 3 grabs the focus immediately after you navigate to them, which is a bit disorienting. Ideally, we would announce what the screen is about (like the explanation text above the text fields).

When you pick a domain name and proceed to the next screen, you see the progress, and it's immediately announced with "We are creating your site". That's awesome! But when the site is created, and you transition to the next screen, nothing happens. Maybe we can announce "Your site has been created!" label?

Another minor thing - maybe we could announce the container with site preview? Currently, it's hard to tell that it's a preview.

**Comment by malinajirka** [comment]
Thanks @khaykov! Definitely great suggestions!! I've tried to fix the issues. I'm not sure it's the best approach, so let me know if you can come up with something better. Thanks!

File: WordPress/src/main/res/layout/new_site_creation_site_info_screen.xml [hunk 1]
```diff
@@ -23,32 +23,40 @@
             android:layout_height="wrap_content"
             android:layout_gravity="center_horizontal">
 
-            <TextView
```
Comment (malinajirka) [inline WordPress/src/main/res/layout/new_site_creation_site_info_screen.xml]: This looks like a big change, but I just wrapped title and subtitle with a linear layout.

**Comment by oguzkocer** [comment]
@malinajirka @khaykov I've updated the base branch to `develop` since we merged everything in. There are some merge conflicts from all the recent changes that needs to be resolved before it can be merged. Let me know if I can help with them!

**Pull request merged**